ixmp/reporting/__init__.py

```diff
@@ -0,0 +1,509 @@
+            zip(repeat('var'), sorted(scenario.var_list())),
+        )
+
+        for ix_type, name in quantities:
+            # List of computations for the quantity and its partial sums
+            comps = list(keys_for_quantity(ix_type, name, scenario))
+
+            # Add to the graph
+            rep.graph.update(comps)
+
+            # Add first 1 or 2 (equ marginals) keys to the list of all
+            # quantities
+            full_comps = comps[:(2 if ix_type == 'equ' else 1)]
+            all_keys.extend(c[0] for c in full_comps)
+
+            # Index the variable name
+            rep._index[name] = full_comps[0][0]
+
+        # Add a key which simply collects all quantities
+        rep.add('all', sorted(all_keys))
+
+        # Add sets
+        for name in scenario.set_list():
+            elements = scenario.set(name)
+            try:
+                elements = elements.tolist()
+            except AttributeError:
+                # pd.DataFrame for a multidimensional set; store as-is
+                pass
+            rep.add(name, elements)
+
+        # Add the scenario itself
+        rep.add('scenario', scenario)
+
+        return rep
+
+    def read_config(self, path):
+        """Configure the Reporter with information from a YAML file at *path*.
+
+        See :meth:`configure`.
+        """
+        path = Path(path)
+        with open(path, 'r') as f:
+            self.configure(config_dir=path.parent, **yaml.load(f))
+
+    def configure(self, path=None, **config):
+        """Configure the Reporter.
+
+        Accepts a *path* to a configuration file and/or keyword arguments.
+        Configuration keys loaded from file are replaced by keyword arguments.
+
+        Valid configuration keys include:
+
+        - *default*: the default reporting key; sets :attr:`default_key`.
+        - *files*: a :class:`dict` mapping keys to file paths.
+        - *alias*: a :class:`dict` mapping aliases to original keys.
+
+        Warns
+        -----
+        UserWarning
+            If *config* contains unrecognized keys.
+        """
+        sections = {'default', 'files', 'alias', 'filters'}
+        config = _config_args(path, config, sections=sections)
+
+        config_dir = config.pop('config_dir', '.')
+
+        # Read sections
+
+        # Default report
+        try:
+            self.default_key = config['default']
+        except KeyError:
+            pass
+
+        # Files with exogenous data
+        for key, path in config.get('files', {}).items():
+            path = Path(path)
+            if not path.is_absolute():
+                # Resolve relative paths relative to the directory containing
+                # the configuration file
+                path = config_dir / path
+            self.add_file(path, key)
+
+        # Aliases
+        for alias, original in config.get('alias', {}).items():
+            self.add(alias, original)
+
+        # Filters
+        self.filter(**config.pop('filters', {}))
```

Comment:
I think this should be `self.filters` for a) consistency with kwarg name and rest of code base and b) `filter` is a keyword in python, so I've tried to stay away from using it explicitly

Reply:
I'll go with `set_filters`, since method names should be verbs.